truststore_java.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"hash"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+var (
+	hasJava    bool
+	hasKeytool bool
+
+	javaHome    string
+	cacertsPath string
+	keytoolPath string
+	storePass   string = "changeit"
+)
+
+func init() {
+	if v := os.Getenv("JAVA_HOME"); v != "" {
+		hasJava = true
+		javaHome = v
+
+		_, err := os.Stat(path.Join(v, "bin/keytool"))
+		if err == nil {
+			hasKeytool = true
+			keytoolPath = path.Join(v, "bin/keytool")
+		}
+
+		cacertsPath = path.Join(v, "jre/lib/security/cacerts")
+	}
+}
+
+func (m *mkcert) checkJava() bool {
+	// exists returns true if the given x509.Certificate's fingerprint
+	// is in the keytool -list output
+	exists := func(c *x509.Certificate, h hash.Hash, keytoolOutput []byte) bool {
+		_, err := h.Write(c.Raw)
+		if err != nil {
+			return false
+		}
+		// keytool outputs SHA1 and SHA256 certificates in uppercase hex
+		// with each octet pair deliniated by ":"
+		fp := []byte(strings.ToUpper(hex.EncodeToString(h.Sum(nil))))
+		return bytes.Contains(keytoolOutput, fp)
+	}
+
+	keytoolOutput, err := exec.Command(keytoolPath, "-list", "-keystore", cacertsPath, "-storepass", storePass).CombinedOutput()
+	if err != nil {
+		hasKeytool = false
+		return false
+	}
+	keytoolOutput = bytes.Replace(keytoolOutput, []byte(":"), nil, -1) // drop : in whole thing
+
+	// pre-Java 9 uses SHA1 fingerprints
+	s1, s256 := sha1.New(), sha256.New()
+	return exists(m.caCert, s1, keytoolOutput) || exists(m.caCert, s256, keytoolOutput)
+}
+
+func (m *mkcert) installJava() {
+	tmp, err := ioutil.TempFile("", "mkcert-java")
+	fatalIfErr(err, "failed creating temp file for java install")
+	defer os.Remove(tmp.Name())
+
+	err = ioutil.WriteFile(tmp.Name(), pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: m.caCert.Raw,
+	}), 0644)
+	fatalIfErr(err, "failed to write java cert into temp file")
```

Comment:
Could you use the certificate file on disk? See for example the macOS `security` input.